p2p/exchange.go

```diff
@@ -230,10 +256,17 @@ func (ex *Exchange[H]) Head(ctx context.Context, opts ...header.HeadOption[H]) (
 // thereafter.
 func (ex *Exchange[H]) GetByHeight(ctx context.Context, height uint64) (H, error) {
```

Comment:
we never really use this path btw.

Reply:
doesn't hurt to cover it you mean to remove it